lms/djangoapps/grades/tests/test_models.py

```diff
@@ -238,3 +239,58 @@ def test_update_or_create_grade(self, already_created):
         if already_created:
             self.assertEquals(created_grade.id, updated_grade.id)
             self.assertEquals(created_grade.earned_all, 6)
+
+
+@ddt.ddt
+class PersistentCourseGradesTest(GradesModelTestCase):
+    """
+    Tests the PersistentCourseGrade model.
+    """
+    def setUp(self):
+        super(PersistentCourseGradesTest, self).setUp()
+        self.params = {
+            "user_id": 12345,
+            "course_id": self.course_key,
+            "course_version": "JoeMcEwing",
+            "course_edited_timestamp": "2016-08-01 18:53:24.354741",
+            "percent_grade": "77.7",
+            "letter_grade": "Great job",
+        }
+
+    def test_update(self):
+        created_grade = PersistentCourseGrade.objects.create(**self.params)
+        self.params["percent_grade"] = 88.8
+        self.params["letter_grade"] = "Better job"
+        updated_grade = PersistentCourseGrade.update_or_create_course_grade(**self.params)
+        self.assertEquals(updated_grade.percent_grade, 88.8)
+        self.assertEquals(updated_grade.letter_grade, "Better job")
+        self.assertEqual(created_grade.id, updated_grade.id)
+
+    def test_create(self):
+        self.assertEquals(len(PersistentCourseGrade.objects.values()), 0)
+        updated_grade = PersistentCourseGrade.update_or_create_course_grade(**self.params)
+        for param in self.params.keys():
```

Comment:
Nit: can remove `.keys()` since the default iterator of a `dict` just returns the keys in the `dict`.